lib/components/stampboard.ts

```diff
@@ -0,0 +1,32 @@
+import { length } from "circuit-json"
+import {
+  type CommonComponentProps,
+  commonComponentProps,
+} from "lib/common/layout"
+import { expectTypesMatch } from "lib/typecheck"
+import { z } from "zod"
+
+export interface StampboardProps extends CommonComponentProps {
+  width?: string | number
+  height?: string | number
+  leftPins?: number
+  rightPins?: number
+  topPins?: number
+  bottomPins?: number
+  pitch?: string | number
+  innerHole?: boolean
+}
+
+export const stampboardProps = commonComponentProps.extend({
+  width: length.optional(),
+  height: length.optional(),
+  leftPins: z.number().optional(),
+  rightPins: z.number().optional(),
+  topPins: z.number().optional(),
+  bottomPins: z.number().optional(),
+  pitch: length.optional(),
+  innerHole: z.boolean().optional(),
```

Comment:
add `leftPinCount`, `leftPins` should be an array of pin labels (do for all pins)

`pitch` should be `pinPitch`